start-from-zero/Javascript/src/api/storeData.js

```diff
@@ -0,0 +1,54 @@
+function FormData(){
+    btn.addEventListener('click',function(){
+        function User(username,password,date,radio,textarea,select){
+        this.username = username;
+        this.password = password;
+        this.date = date;
+        this.radio = radio;
+        this.textarea = textarea;
+        this.select = select.forEach(element=>{
+            this[element.name] = element.value;
+        })
+        }
+
+        const username = document.getElementById('username-id').value;
+        const password = document.getElementById('password-id').value;
+        const calendar = document.getElementById('calendar-id').value;
+        const checkbox = document.querySelectorAll('.input-checkbox');
+        const textarea = document.querySelectorAll('textarea');
+        const legend = document.querySelectorAll('legend');
+        const select = document.querySelectorAll('select');
+        
+        const textareaArr = [];
```

Comment:
I would much rather that we don't couple content with the element index; if I add another textarea you will have to change the whole data grabber process; any better way you can think of?

Reply:
Yes. I solved this.